lightning/src/chain/transaction.rs

```diff
@@ -56,13 +57,9 @@ pub struct OutPoint {
 }
 
 impl OutPoint {
-	/// Convert an `OutPoint` to a lightning channel id.
-	pub fn to_channel_id(&self) -> [u8; 32] {
-		let mut res = [0; 32];
-		res[..].copy_from_slice(&self.txid[..]);
-		res[30] ^= ((self.index >> 8) & 0xff) as u8;
-		res[31] ^= ((self.index >> 0) & 0xff) as u8;
-		res
+	/// Convert an `OutPoint` to a lightning channel id. Alternatively use ChannelId::from_funding_outpoint().
+	pub fn to_channel_id(&self) -> ChannelId {
```

Comment:
This method existed in OutPoint, and AFAIK OutPoint cannot be used to generate other type of ChannelId, so I did not change it now, thought it would make sense.